packages/mgt-chat/src/utils/rewriteEmojiContent.ts

```diff
@@ -1,32 +1,63 @@
 /**
- * Regex to detect and extract emoji alt text
- *
- * Pattern breakdown:
- * (<emoji[^>]+): Captures the opening emoji tag, including any attributes.
- * alt=["'](\w*[^"']*)["']: Matches and captures the "alt" attribute value within single or double quotes. The value can contain word characters but not quotes.
- * (.[^>]): Captures any remaining text within the opening emoji tag, excluding the closing angle bracket.
- * ><\/emoji>: Matches the remaining part of the tag.
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
  */
-const emojiRegex = /(<emoji[^>]+)alt=["'](\w*[^"']*)["'](.[^>]+)><\/emoji>/;
-const emojiMatch = (messageContent: string): RegExpMatchArray | null => {
-  return messageContent.match(emojiRegex);
-};
-// iterative repave the emoji custom element with the content of the alt attribute
-// on the emoji element
-const processEmojiContent = (messageContent: string): string => {
-  let result = messageContent;
-  let match = emojiMatch(result);
-  while (match) {
-    result = result.replace(emojiRegex, '$2');
-    match = emojiMatch(result);
+
+/**
+ * Checks if DOM content has emojis and other HTML content.
+ * @param dom the html content parsed into HTMLDocument.
+ * @param emojisCount number of emojis in the content.
+ * @returns true if only one emoji is in the content without other content, otherwise false.
+ */
+const hasOtherContent = (dom: Document, emojisCount: number): boolean => {
+  const isPtag = dom.body.firstChild?.nodeName === 'P';
```

Comment:
What happens if there are multiple P tags each containing a single emojii?

Say:
```html
<p><emoji id=\"hearteyes\" alt=\"😍\" title=\"Heart eyes\"></emoji></p>
<p><emoji id=\"1f92a_zanyface\" alt=\"🤪\" title=\"Zany face\"></emoji></p>
<p><emoji id=\"cool\" alt=\"😎\" title=\"Cool\"></emoji></p>
```

Reply:
NB: that is the content of an actual message I sent from Teams

Reply:
In this case, we have 3 emojis already so it will short circuit on line 34 (set the emoji size to 20) and not reach this block.

Reply:
That's how it worked but @sebastienlevert requested parity with the Teams experience above https://github.com/microsoftgraph/microsoft-graph-toolkit/pull/2870#issuecomment-1821263282